src/pages/BlogPage.js

```diff
@@ -0,0 +1,75 @@
+import { useEffect, useState } from "react";
+import { useNavigate } from "react-router-dom";
+import axiosInstance from "../api/axios";
+import "./pageStyle/BlogPage.css";
+
+const BlogPage = () => {
+  const [displayBlogs, setDisplayBlogs] = useState([]);
+  const [page, setPage] = useState(1);
+  let navigate = useNavigate();
+
+  const getBlog = async () => {
+    try {
+      const { data } = await axiosInstance.get();
+      const allPost = data.getBlogPost.docs;
+      setDisplayBlogs((prevPost) => [...prevPost, ...allPost]);
+      setPage(page + 1);
+    } catch {
+      console.log("Error occured");
+    }
+  };
+
+  useEffect(() => {
+    const handleScroll = (e) => {
+      if (
+        window.innerHeight + e.target.documentElement.scrollTop + 1 >=
+        e.target.documentElement.scrollHeight
+      ) {
+        getBlog();
+      }
+    };
+    getBlog();
+    window.addEventListener("scroll", handleScroll);
+  }, []);
```

Comment:
This should have a `cleanup` function

Reply:
now added by removing the event on window